internal/rest/avi_obj_vs.go

```diff
@@ -371,38 +371,66 @@ func AviVsHttpPSAdd(vs_meta interface{}, isEVH bool) []*avimodels.HTTPPolicies {
 func (rest *RestOperations) StatusUpdate(rest_op *utils.RestOp, vs_cache_obj *avicache.AviVsCache, svc_mdata_obj *avicache.ServiceMetadataObj, parentVsObj *avicache.AviVsCache, key string, oldObj bool) error {
 	if oldObj {
 		if rest_op.Method == utils.RestPost || rest_op.Method == utils.RestDelete || rest_op.Method == utils.RestPut {
-			for _, poolkey := range vs_cache_obj.PoolKeyCollection {
-				// Fetch the pool object from cache and check the service metadata
-				pool_cache, ok := rest.cache.PoolCache.AviCacheGet(poolkey)
-				if ok {
-					utils.AviLog.Infof("key: %s, msg: found pool: %s, will update status", key, poolkey.Name)
-					pool_cache_obj, found := pool_cache.(*avicache.AviPoolCache)
-					if found {
-						IPAddrs := rest.GetIPAddrsFromCache(vs_cache_obj)
-						if len(pool_cache_obj.ServiceMetadataObj.NamespaceServiceName) > 0 {
-							updateOptions := status.UpdateOptions{
-								Vip:                IPAddrs,
-								ServiceMetadata:    pool_cache_obj.ServiceMetadataObj,
-								Key:                key,
-								VirtualServiceUUID: vs_cache_obj.Uuid,
-								VSName:             vs_cache_obj.Name,
-							}
-							statusOption := status.StatusOptions{
-								ObjType: utils.L4LBService,
-								Op:      lib.UpdateStatus,
-								Options: &updateOptions,
-							}
-							status.PublishToStatusQueue(updateOptions.ServiceMetadata.NamespaceServiceName[0], statusOption)
-						} else if pool_cache_obj.ServiceMetadataObj.Namespace != "" {
-							if len(IPAddrs) == 0 {
-								IPAddrs = []string{parentVsObj.Vip}
+			//in secure case, PoolKeyCollection will be empty
+			if len(vs_cache_obj.PoolKeyCollection) > 0 {
```

Comment:
I think this might not be true? In case of secure VSes, the insecure hosts might land up in the pool key collection, in case of SNI model. I see that we are processing SNIChildCollection *only* when PoolKeyCollection has no elements which doesn't seem to be right.